packages/grafana-ui/src/components/DataSourceSettings/types.ts

```diff
@@ -2,8 +2,20 @@ import React from 'react';
 import { DataSourceSettings } from '@grafana/data';
 
 export interface AzureAuthSettings {
-  azureAuthEnabled: boolean;
-  azureSettingsUI?: React.ComponentType<HttpSettingsBaseProps>;
+  /** Set to true if Azure authentication supported by the datasource */
+  readonly azureAuthSupported: boolean;
```

Comment:
I think this initiative is great and I agree that we need to improve the code according to your description in the PR.

However, this is a breaking change and will make plugins using these APIs to stop functioning until being upgraded to use this version of `@grafana/ui`. 

Is this used only by bundled plugins or do we have any external plugins using this? @grafana/cloud-datasources do you know?

Reply:
Not that I know of, but theoretically it can be used by any data source that enable this feature. I know for sure that the only other Azure data source that we maintain - Azure data explorer - authenticates directly against Azure so it's not using the http client provider. @kostrse do you know if any other data sources that might be using `Azure auth` through DataSourceHttpSettings?

Reply:
This has been used only for Prometheus datasource and behind a feature flag (`prometheus_azure_auth`).
So, while it's a change to public interface, there should not be any impact.